zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,870 @@
+
+For full documentation, see ``help(from_blaze)`` or ``from_blaze?`` in jupyter.
+
+Using our Pipeline DataSets and Columns
+---------------------------------------
+
+Once we have mapped our blaze expressions into Pipeline API objects, we may
+use them just like any other datasets or columns. For more information on how
+to run a pipeline or using the Pipeline API, see:
+www.quantopian.com/help#pipeline-api
+"""
+from __future__ import division, absolute_import
+
+from abc import ABCMeta, abstractproperty
+from collections import namedtuple, defaultdict
+from itertools import count
+import warnings
+from weakref import WeakKeyDictionary
+
+import blaze as bz
+from datashape import (
+    Date,
+    DateTime,
+    Option,
+    float64,
+    isrecord,
+    isscalar,
+    promote,
+)
+from numpy.lib.stride_tricks import as_strided
+from odo import odo
+import pandas as pd
+from toolz import (
+    complement,
+    compose,
+    concat,
+    flip,
+    groupby,
+    identity,
+    memoize,
+)
+import toolz.curried.operator as op
+from six import with_metaclass, PY2, iteritems
+
+
+from ..data.dataset import DataSet, Column
+from zipline.lib.adjusted_array import adjusted_array
+from zipline.lib.adjustment import Float64Overwrite
+
+
+AD_FIELD_NAME = 'asof_date'
+TS_FIELD_NAME = 'timestamp'
+SID_FIELD_NAME = 'sid'
+valid_deltas_node_types = (
+    bz.expr.Field,
+    bz.expr.ReLabel,
+    bz.expr.Symbol,
+)
+traversable_nodes = (
+    bz.expr.Field,
+    bz.expr.Label,
+)
+is_invalid_deltas_node = complement(flip(isinstance, valid_deltas_node_types))
+getname = op.attrgetter('__name__')
+
+
+class _ExprRepr(object):
+    """Box for repring expressions with the str of the expression.
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression to box for repring.
+    """
+    __slots__ = 'expr',
+
+    def __init__(self, expr):
+        self.expr = expr
+
+    def __repr__(self):
+        return str(self.expr)
+    __str__ = __repr__
+
+
+class ExprData(namedtuple('ExprData', 'expr deltas resources')):
+    """A pair of expressions and a data resources.
+
+    Parameters
+    ----------
+    expr : Expr
+        The first known values.
+    deltas : Expr, optional
+        The deltas for the data.
+    resources : resource or dict of resources, optional
+        The resources to compute the exprs against.
+    """
+    def __new__(cls, expr, deltas=None, resources=None):
+        return super(ExprData, cls).__new__(cls, expr, deltas, resources)
+
+    def __repr__(self):
+        # If the expressions have _resources() then the repr will
+        # drive computation so we box them.
+        cls = type(self)
+        return super(ExprData, cls).__repr__(cls(
+            _ExprRepr(self.expr),
+            _ExprRepr(self.deltas),
+            self.resources,
+        ))
+
+
+class InvalidField(with_metaclass(ABCMeta)):
+    """A field that raises an exception that indicates that the
```

Comment:
English style nit: "exception indicating that" means the same thing and doesn't repeat "that" twice in three words.